### ISSUE TITLE
Add model checkpointing: periodic, last, best loss, best accuracy

### ISSUE THREAD

**Issue by Ad7amstein** [Ad7amstein/deep-activity-recognition#30]
Implement model checkpointing during training to improve reproducibility and recovery.
- Save a checkpoint every n=2 epochs.
- Save the model from the last epoch.
- Save the model with the best validation loss.
- Save the model with the best validation accuracy.
